dev/tests/integration/testsuite/Magento/Sales/_files/order_list.php

```diff
@@ -48,16 +49,30 @@
     ],
 ];
 
+/** @var OrderRepositoryInterface $orderRepository */
+$orderRepository = $objectManager->create(OrderRepositoryInterface::class);
 /** @var array $orderData */
 foreach ($orders as $orderData) {
     /** @var $order \Magento\Sales\Model\Order */
     $order = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
         \Magento\Sales\Model\Order::class
     );
+
+    // Reset addresses
+    /** @var Order\Address $billingAddress */
+    $billingAddress = $objectManager->create(OrderAddress::class, ['data' => $addressData]);
+    $billingAddress->setAddressType('billing');
+
+    $shippingAddress = clone $billingAddress;
+    $shippingAddress->setId(null)->setAddressType('shipping');
+
     $order
         ->setData($orderData)
         ->addItem($orderItem)
+        ->setCustomerIsGuest(true)
+        ->setCustomerEmail('customer@null.com')
```

Comment:
@evktalo so we had some issues with an API test because the order needed a valid email attached to it to save. I also refactored the code a bit to use the `$orderRepository` when saving. This makes it make the file `dev/tests/integration/testsuite/Magento/Sales/_files/order.php`